internal/repos/jvm_packages.go

```diff
@@ -72,6 +151,30 @@ func (s *JVMPackagesSource) GetRepo(ctx context.Context, artifactPath string) (*
 		return nil, err
 	}
 
+	dbDeps, err := s.dbStore.GetJVMDependencyRepos(ctx, dbstore.GetJVMDependencyReposOpts{
+		ArtifactName: artifactPath,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "dbstore.GetJVMDependencyRepos")
+	}
+
+	for _, dep := range dbDeps {
+		parsedModule, err := reposource.ParseMavenModule(dep.Module)
+		if err != nil {
+			log15.Warn("error parsing maven module", "error", err, "module", dep.Module)
+			continue
+		}
+		fullDependencyString := reposource.MavenDependency{
```

Comment:
You already have a `MavenDependency`, the `ParseMavenDependency(_.CoursierSyntax())` roundtrip is unnecessary.
```suggestion
		dependency := reposource.MavenDependency{
			MavenModule: parsedModule,
			Version:     dep.Version,
		}
```